coderd/database/dbauthz/dbauthz.go

```diff
@@ -2084,13 +2114,13 @@ func (q *querier) GetProvisionerJobByID(ctx context.Context, id uuid.UUID) (data
 		// can read the job.
 		_, err := q.GetWorkspaceBuildByJobID(ctx, id)
 		if err != nil {
-			return database.ProvisionerJob{}, err
+			return database.ProvisionerJob{}, xerrors.Errorf("fetch related workspace build: %w", err)
```

Comment:
Self review: not related to our changes, but was useful to troubleshooting and will likely be again.